primitives/core/src/lib.rs

```diff
@@ -72,6 +71,42 @@ impl From<MessageSendError> for &'static str {
 	}
 }
 
+/// The origin of an inbound message.
+#[derive(Encode, Decode, MaxEncodedLen, Clone, Eq, PartialEq, TypeInfo, Debug)]
+pub enum AggregateMessageOrigin {
```

Comment:
Just an FYI, In the BridgeHub system parachain, we are also wanting to use the message queue for buffering messages before bridging them to Ethereum. Busy implementing this in https://github.com/Snowfork/snowbridge/pull/851.

In our case, this our aggregate origin looks like this:
```
pub enum AggregateMessageOrigin {
 	#[codec(index = 0)]
 	Parachain(ParaId),
 }
```

So each parachain sending messages to ethereum has its own queue.

Hoping we can add an enum variant to cater for this use case.

Reply:
Also just a note to myself (for future reference):

For the BridgeHub system parachain, we'd like for the message-queue pallet to be registered after our Ethereum-specific pallets, to ensure it's on_initialize hander is called after our on_initialize handlers.

This is to ensure our message processing queues are correctly initialised before message-queue starts doing its thing.

Reply:
Nice. Yea LMK if you have feedback on the MQ pallet. 

Currently we use the MQ pallet for inbound dispatch queues (not delivery) like DMP/XCMP on paras and UMP dispatch on the relay. But I guess it can also be used for delivery, so we could add an `OutboundSnowbridged(ParaId)` or similar.  

How do you currently handle incoming snowbridged messages? Are they executed immediately or queued as well?

Reply:
Excuse the tardy reply, I was OOO for most of this week.

> Currently we use the MQ pallet for inbound dispatch queues (not delivery) like DMP/XCMP on paras and UMP on the relay. 

This is fine and I still think still fits the design envisioned for MessageQueue. Strictly speaking, in the Polkadot->Ethereum direction, when messages from origin parachains reach bridge hub, they certainly haven't been finally delivered yet, and still need to be processed before they are suitable for delivery to Ethereum. Buffering is needed, as our bridge can only process a finite number of messages per block, in order for the weights/benchmarking side to work out.

> How do you currently handle incoming snowbridged messages? Are they executed immediately or queued as well?

Inbound messages from ethereum are immediately routed to a sibling parachain via HRMP - no queuing or buffering or anything like that. For reference, see https://github.com/Snowfork/snowbridge/blob/bd1773d2e983e3c558e61db5b4348ecc5fe922c4/parachain/pallets/inbound-queue/src/lib.rs#L199.

So yeah, is standard XCMP.

> PS: We can also simplify the traits in the parachain-system pallet, as to not force every parachain to adopt the same AggregateMessageOrigin.

Yes please, our enum variant in `AggregateMessageOrigin` very specific and probably won't make much sense for parachains other than BridgeHub.